models/importCsvData.js

```diff
@@ -0,0 +1,265 @@
+        const d = new Date(item);
+        if(!isNaN(d.getTime())) card.createdAt = d;
+      } else if (i === index.updatedAt) {
+        const d = new Date(item);
+        if(!isNaN(d.getTime())) card.dateLastActivity = d;
+      } else if (i === index.state) {
+        const wantedList = Lists.findOne({
+          boardId: this.board._id,
+          archived: false,
+          title: item,
+        });
+        if (wantedList) {
+          card.listId = wantedList._id;
+        }
+      } else if (i === index.manHour) {
+        const n = parseInt(item, 10);
+        if(n > 0) card.manHour = n;
+      } else if (i === index.dueDate) {
+        const d = new Date(`${item} 12:00:00`);
+        if(!isNaN(d.getTime())) card.dueDate = d;
+      } else if (i === index.startDate) {
+        const d = new Date(`${item} 12:00:00`);
+        if(!isNaN(d.getTime())) card.startDate = d;
+      } else if (i === index.finishDate) {
+        const d = new Date(`${item} 12:00:00`);
+        if(!isNaN(d.getTime())) card.finishDate = d;
+      }
+    }
+
+    return card.title ? Cards.insert(card) : null;
+  }
+
+  importCards(rows, listId, sortIndex) {
+    this.list = Lists.findOne(listId);
+    this.board = this.list.board();
+    this.board.memberUsers().forEach((user) => {
+      this.mapName2Id[ user.username ] = user._id;
+    });
+
+    if (rows.length > 0) {
+      this.findIndex(rows.shift());
+      const index = this.mapIndex;
+      if (index.title >= 0) {
+        const cardIds = [];
+        rows.forEach((row) => {
+          const cardId = this.insertRowAsCard(row, sortIndex++);
+          if (cardId) cardIds.push(cardId);
+        });
+        return cardIds;
+      }
+    }
+
+    throw new Meteor.Error('error-json-schema');
+  }
+
+  exportCards(filter, boardId, tsv) {
+    const cards = Cards.find(filter, { sort: ['listId', 'sort'] }).fetch();
+
+    const separator = tsv ? '\t' : ',';
+    const board = Boards.findOne(boardId);
+    const users = board.memberUsers().fetch();
+    const lists = board.lists().fetch();
+    const mapuser = {};
+    const maplist = {};
+    const maplabel = {};
+    users.forEach((user) => {
+      mapuser[user._id] = user.username;
+    });
+    lists.forEach((list) => {
+      maplist[list._id] = list.title;
+    });
+    board.labels.forEach((label) => {
+      maplabel[label._id] = label.name || label.color;
+    });
+
+    let maxMembers = 1;
+    let maxLabels = 1;
+    cards.forEach((card) => {
+      if(card.members && card.members.length > maxMembers) maxMembers = card.members.length;
+      if(card.labelIds && card.labelIds.length > maxLabels) maxLabels = card.labelIds.length;
+    });
+    const hmembers = _.range(maxMembers).map(() => 'member');
+    const hlabels = _.range(maxLabels).map(() => 'label');
+    const headers = ['title',
+                     'description',
+                     'status',
+                     'owner',
+                     ].concat(
```

Comment:
As number of members and labels may be different for cards, we need use max of them as columns, while set those with less members or labels with empty cells.